src/test/java/fr/insee/publicenemy/api/infrastructure/queen/dto/SurveyUnitSerializerTest.java

```diff
@@ -0,0 +1,139 @@
+        attributes.put("att3", listValue);
+
+        SurveyUnitData data = new SurveyUnitData(attributes);
+
+        surveyUnits.add(new SurveyUnitDto("1", "q1", data, SurveyUnitStateData.createInitialStateData()));
+        surveyUnits.add(new SurveyUnitDto("2", "q2", data, SurveyUnitStateData.createInitialStateData()));
+
+        ObjectMapper mapper = new ObjectMapper();
+
+        SimpleModule module = new SimpleModule();
+        module.addSerializer(SurveyUnitDto.class, new SurveyUnitSerializer());
+        mapper.registerModule(module);
+
+        String jsonSurveyUnits = mapper.writeValueAsString(surveyUnits);
+
+        assertEquals("""
+        [
+           {
+              "id":"1",
+              "questionnaireId":"q1",
+              "personalization":[
+                 {
+                    "name":"whoAnswers1",
+                    "value":"MrDupond"
+                 },
+                 {
+                    "name":"whoAnswers2",
+                    "value":""
+                 },
+                 {
+                    "name":"whoAnswers3",
+                    "value":""
+                 }
+              ],
+              "comment":{
+              },
+              "data":{
+                 "EXTERNAL":{
+                    "att1":"1",
+                    "att2":"CS70058",
+                    "att3":[
+                       "value1",
+                       "value2",
+                       "value3",
+                       "value4",
+                       "value5",
+                       "value6"
+                    ]
+                 }
+              },
+              "stateData":{
+                 "currentPage":"0",
+                 "date":900000000,
+                 "state":"INIT"
+              }
+           },
+           {
+              "id":"2",
+              "questionnaireId":"q2",
+              "personalization":[
+                 {
+                    "name":"whoAnswers1",
+                    "value":"MrDupond"
+                 },
+                 {
+                    "name":"whoAnswers2",
+                    "value":""
+                 },
+                 {
+                    "name":"whoAnswers3",
+                    "value":""
+                 }
+              ],
+              "comment":{
+              },
+              "data":{
+                 "EXTERNAL":{
+                    "att1":"1",
+                    "att2":"CS70058",
+                    "att3":[
+                       "value1",
+                       "value2",
+                       "value3",
+                       "value4",
+                       "value5",
+                       "value6"
+                    ]
+                 }
+              },
+              "stateData":{
+                 "currentPage":"0",
+                 "date":900000000,
+                 "state":"INIT"
+              }
+           }
+        ]""".replaceAll("\\s+","")
+                ,jsonSurveyUnits.replaceAll("\\s+",""));
```

Comment:
neat trick!

Reply:
too bad it replaces spaces in data too...

Reply:
There might be a library to test differences between jsons maybe?